pkg/topology/mock/mock.go

```diff
@@ -164,7 +164,7 @@ func (m *mock) IsWithinDepth(addr swarm.Address) bool {
 	if m.isWithinFunc != nil {
 		return m.isWithinFunc(addr)
 	}
-	return false
+	return true
```

Comment:
@acud The reason why the default is now `true` is because the chunk must be in the neighborhood of the peer with this change. The majority of the tests expect a valid receipt so the closest peers in the tests must be in the  neighborhood. As  of writing this, there is only one test that expects a peer to be out of depth.

Reply:
In short, the previous default `false` causes every test to fail.

Reply:
@istae then can we have an explicit change to the tests, in which a custom mocked `IsWithinFunc` is injected with a `true`? my concern is that with this change, also forwarder nodes (and there are forwarding tests) will see the chunk as within depth, leading to a change in the test cases' final state

Reply:
it is also counter-intuitive that we would now test everything with the following assumptions by default:
- originator node sees all chunks as within neighborhood
- fowarder node sees all chunks as within neighborhood
- storer node sees all chunks as within neighborhood
- replicating node sees all chunks as within neighborhood

a bit too permissive IMO. so I would vote for reverting this back to `false` and have an explicit `true` injected with a mock wherever needed